src/sentry/spans/span.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+import hashlib
+from dataclasses import dataclass
+from typing import Mapping, Optional, TypedDict
+
+from sentry import nodestore
+
+SPAN_SCHEMA_VERSION = 1
+
+
+class SpanData(TypedDict):
+    attributes: Mapping[str, object]
+    id: str
+    project_id: int
+    tags: Mapping[str, str]
+    trace_id: str
+
+
+@dataclass(frozen=True)
+class Span:
+    attributes: Mapping[str, object]
+    id: str
+    project_id: int
+    tags: Mapping[str, str]
+    trace_id: str
+
+    def to_dict(self) -> SpanData:
+        return {
+            "attributes": self.attributes,
+            "id": self.id,
+            "project_id": self.project_id,
+            "tags": self.tags,
+            "trace_id": self.trace_id,
+        }
+
+    @classmethod
+    def from_dict(cls, data: SpanData) -> Span:
+        return cls(
+            data["attributes"],
+            data["id"],
+            data["project_id"],
+            data["tags"],
+            data["trace_id"],
+        )
+
+    @classmethod
+    def key(cls, version: int, project_id: int, trace_id: str, span_id: str) -> str:
+        identifier = hashlib.md5(f"{project_id}:{trace_id}:{span_id}".encode()).hexdigest()
+        return f"s:{version}:{identifier}"
```

Comment:
My suggestion was the below. Doing this it would be easy to do prefix lookups from sentry directly.
```suggestion
        return f"s:{version}:{project_id}:{trace_id}:{span_id}"
```